src/utils.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+let unnamedEditorCount = 0;
+
+export function normalizeArguments( args ) {
+	const normalized = {
+		editors: {}
+	};
+
+	// attach( editor );
+	// attach( { foo: editor1, bar: editor2 } );
+	if ( args.length === 1 ) {
+		const editorOrEditorPairs = args[ 0 ];
+
+		// attach( editor );
+		if ( isEditorInstance( editorOrEditorPairs ) ) {
+			normalized.editors[ getNextEditorName() ] = args[ 0 ];
+		}
+		// attach( { foo: editor1, bar: editor2, ... } );
+		else {
+			for ( const name in editorOrEditorPairs ) {
+				normalized.editors[ name ] = editorOrEditorPairs[ name ];
+			}
+		}
+	}
+	// attach( editor, { options } );
+	// attach( { foo: editor1, bar: editor2 }, { options } );
+	else {
```

Comment:
This duplicates the above `if` in 90%, the only difference I see is `normalized.options = args[ 1 ];`

Reply:
A nice catch! I simplified things and added a support for a deprecated syntax with a warning.